crossbar/router/auth/pending.py

```diff
@@ -45,27 +72,28 @@ class PendingAuth:
 
     AUTHMETHOD = 'abstract'
 
-    def __init__(self, session, config):
+    def __init__(self, pending_session_id, transport_info, realm_container, config):
         """
+        :param int pending_session_id: the Session ID if this succeeds
+
+        :param dict transport_info: information about the session's rtansport
+
+        :param IRealmContainer realm_container: access configured realms / roles
 
-        :param session: The authenticating session.
-        :type session: obj
         :param config: Authentication configuration to apply for the pending auth.
         :type config: dict
         """
-        # The session that is authenticating
-        self._session = session
 
         # Details about the authenticating session
         self._session_details = {
-            'transport': session._transport._transport_info,
-            'session': session._pending_session_id,
+            'transport': transport_info,
```

Comment:
I think this makes a lot of sense: the job of collecting infos for a transport should not be done in authenticators, but the stuff that creates and calls authenticators - IOW, exactly as this change does: provide it from outside. fwiw, this seamingly harmless "transport info" thing has been haunting us in different places;) this cleans up one of those